schemas/dns.yml

```diff
@@ -0,0 +1,210 @@
+        create dns events of type `dns.type:query`.
+
+        If your source of DNS events gives you answers as well, you should create
+        one event per query (optionally as soon as the query is seen). And a
+        second event containing all query details as well as an array of answers.
+      example: answer
+
+    - name: id
+      level: extended
+      type: keyword
+      description: >
+        The DNS packet identifier assigned by the program that generated the
+        query. The identifier is copied to the response.
+      example: 62111
+
+    - name: op_code
+      level: extended
+      type: keyword
+      description: >
+        The DNS operation code that specifies the kind of query in the message.
+        This value is set by the originator of a query and copied into the
+        response.
+      example: QUERY
+
+    - name: flags
+      level: extended
+      type: keyword
+      short: Array of DNS flags.
+      description: >
+        Array of 2 letter DNS flags.
+
+        Expected values are: AA, TC, RD, RA, AD, CD, DO
+      example: [RD, RA]
+
+    - name: response_code
+      level: extended
+      type: keyword
+      description: The DNS response code.
+      example: NOERROR
+
+    - name: question.name
+      level: extended
+      type: keyword
+      short: The name being queried.
+      description: >
+        The name being queried.
+
+        If the name field contains non-printable
+        characters (below 32 or above 126), those characters should be represented
+        as escaped base 10 integers (\DDD). Back slashes and quotes should be escaped.
+        Tabs, carriage returns, and line feeds should be converted to \t, \r, and
+        \n respectively.
+      example: www.google.com
+
+    - name: question.type
+      level: extended
+      type: keyword
+      description: The type of record being queried.
+      example: AAAA
+
+    - name: question.class
+      level: extended
+      type: keyword
+      description: The class of of records being queried.
+      example: IN
+
+    - name: question.registered_domain
+      level: extended
+      type: keyword
+      short: The highest registered domain, stripped of the subdomain.
+      description: >
+        The highest registered domain, stripped of the subdomain.
+
+        For example, the registered domain for "foo.google.com" is "google.com".
+
+        This value can be determined precisely with a list like the public
+        suffix list (http://publicsuffix.org). Trying to approximate this by
+        simply taking the last two labels will not work well for TLDs such as "co.uk".
+      example: google.com
+
+    - name: question.length
+      level: extended
+      type: long
+      short: The length of the name being queried (in bytes).
+      description: >
+        The length of the name being queried (in bytes).
+
+        This is useful to detect data exfiltration via DNS queries.
+      example: 42
+
+    - name: answers_count
```

Comment:
Similar to my comment on `question.length` above, the number of entries in an array is fairly easy to calculate with a scripted field, so why include it explicitly?

E.g.:
```
  "script_fields": {
    "dns.answers_count": {
      "script": {
        "source": "doc['dns.ans.data'].size()"
      }
    }
  }
```

Reply:
The scripted field is just as good for display purposes, you're right.

However is it common practice to filter on or aggregate on answer count? Not sure I see a clear use case...

I agree we can take this out for now. We can re-add if someone chimes in with a concrete use case for it.